roles/postgresql/tasks/main.yml

```diff
@@ -0,0 +1,86 @@
+---
+- name: Create PostgreSQL storage directory
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+    mode: "0755"
+    owner: root
+    group: root
+  loop:
+    - "{{ postgresql_data_dir }}"
+    - /etc/postgresql
+
+- name: Pull PostgreSQL container image
+  containers.podman.podman_image:
+    name: "{{ postgresql_image }}"
+    state: present
+
+- name: Configure host based authentication
+  template:
+    src: pg_hba.conf.j2
+    dest: /etc/postgresql/pg_hba.conf
+    mode: 0644
+
+- name: Initialize databases and users
+  template:
+    src: init.sh.j2
+    dest: /tmp/postgresql_init.sh
+    mode: 0755
+
+- name: Create PostgreSQL secrets
+  containers.podman.podman_secret:
+    name: postgresql-admin-password
+    data: "{{ postgresql_admin_password }}"
+    labels:
+      app: postgresql
+
+- name: Check for existing database
+  stat:
+    path: "{{ postgresql_data_dir }}/PG_VERSION"
+  register: db_initialized
+
+- name: Deploy PostgreSQL container
+  containers.podman.podman_container:
+    name: "{{ postgresql_container_name }}"
+    image: "{{ postgresql_image }}"
+    state: quadlet
+    command:
+      - "postgres"
+      - "-c"
+      - "config_file=/etc/postgresql/postgresql.conf"
```

Comment:
I am surprised we have to specify this, as postgresql on EL should have sane defaults set.

Reply:
Yeah, now I changed the way I did before, made it so complex earlier which wasn't necessary do dropped it for now.